connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -337,6 +330,71 @@ object KustoWriter {
     CsvRowResult(fields.toArray, size + fields.size)
   }
 
+  private def convertOneField(row: InternalRow, schemaFields: Array[StructField], fieldIndexInRow: Int, timeZone: String) = {
+    val dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", TimeZone.getTimeZone(timeZone))
+    val dataType = schemaFields(fieldIndexInRow).dataType
+    dataType match {
+      case DateType => DateTimeUtils.toJavaDate(row.getInt(fieldIndexInRow)).toString
+      case TimestampType => dateFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(fieldIndexInRow)))
+      case StringType => row.getString(fieldIndexInRow)
+      case BooleanType => row.getBoolean(fieldIndexInRow).toString
+      case _ : StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], timeZone)
+      case _ : ArrayType => ConvertArrayToCsv(row.getArray(fieldIndexInRow), dataType, timeZone)
+      case _ => row.get(fieldIndexInRow, dataType).toString
+    }
+  }
+
+  private def convertStructToCsv(row: InternalRow, schema: StructType, timeZone: String): String = {
```

Comment:
why is this called `convertStructToCsv` whereas `convertOneField` is missing the csv postfix?

Reply:
convertOneField no longer exists.